app/src/main/java/sopt/motivoo/util/custom_view/MotivooMyPieChart.kt

```diff
@@ -0,0 +1,131 @@
+package sopt.motivoo.util.custom_view
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.RectF
+import android.util.AttributeSet
+import android.view.View
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.toBitmap
+import sopt.motivoo.R
+import sopt.motivoo.util.extension.px
+import kotlin.math.abs
+import kotlin.math.cos
+import kotlin.math.sin
+
+class MotivooMyPieChart @JvmOverloads constructor(
+    context: Context,
+    attributeSet: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : View(context, attributeSet, defStyleAttr) {
+    private var stepCount = 0f
+
+    private val progressBarMyPaint: Paint = Paint()
+
+    private val centerCirclePaint: Paint = Paint()
+
+    private val drawRectArea =
+        RectF(LAYOUT_SPACING.px, LAYOUT_SPACING.px, DRAW_AREA_SIZE.px, DRAW_AREA_SIZE.px)
+
+    private var myImage: Bitmap?
+
+    private val layoutSize = LAYOUT_SIZE.px.toInt()
+
+    init {
+        myImage = ContextCompat.getDrawable(context, R.drawable.ic_child_user)?.run {
+            toBitmap(IMAGE_SIZE.px.toInt(), IMAGE_SIZE.px.toInt())
+        }
+
+        context.theme.obtainStyledAttributes(
+            attributeSet, R.styleable.MotivooPieChart, defStyleAttr, defStyleAttr
+        ).run {
+            progressBarMyPaint.run {
+                color = getColor(R.styleable.MotivooPieChart_progressMyColor, Color.BLUE)
+                style = Paint.Style.STROKE
+                strokeWidth = 9.px
+            }
+            centerCirclePaint.run {
+                color = getColor(R.styleable.MotivooPieChart_centerCicleColor, Color.BLACK)
+                style = Paint.Style.STROKE
+                strokeWidth = 3.px
+            }
+            recycle()
+        }
```

Comment:
run이 중복 사용되어 어떤 수신자를 사용하는지 모호한 경우가 존재한 것 같습니다! 둘을 다른 스코프 함수를 사용해보면 어떨까요 !?

또한 [스코프 함수 공식문서](https://kotlinlang.org/docs/scope-functions.html#distinctions)를 보면, 과도하게 많은 사용은 주의가 필요하다고 나와있습니다.

_Although scope functions can make your code more concise, avoid overusing them: it can make your code hard to read and lead to errors. We also recommend that you avoid nesting scope functions and be careful when chaining them because it's easy to get confused about the current context object and value of this or it._

이를 고려하면 스코프 함수를 좀 더 잘 활용할 수 있을 것 같습니다 !

Reply:
[Feedback commit : 371651b](https://github.com/Team-Motivoo/Motivoo-Android/pull/35/commits/371651b36e88e8275b0f49713c426e6d1b1007ad) 좋은 피드백 감사합니다 !!